openedx/core/djangoapps/credit/models.py

```diff
@@ -186,8 +196,141 @@ class CreditEligibility(TimeStampedModel):
+    """
+    A request for credit from a particular credit provider.
+
+    When a user initiates a request for credit, a CreditRequest record will be created.
+    Each CreditRequest is assigned a unique identifier so we can find it when the request
+    is approved by the provider.  The CreditRequest record stores the parameters to be sent
+    at the time the request is made.  If the user re-issues the request
+    (perhaps because the user did not finish filling in forms on the credit provider's site),
+    the request record will be updated, but the UUID will remain the same.
+    """
+
+    uuid = models.CharField(max_length=32, unique=True, db_index=True)
+    username = models.CharField(max_length=255, db_index=True)
+    course = models.ForeignKey(CreditCourse, related_name="credit_requests")
+    provider = models.ForeignKey(CreditProvider, related_name="credit_requests")
+    parameters = JSONField()
+
+    def current_status(self):  # pylint: disable=no-member
         """
-        Model metadata.
+        Retrieve the current status for a request.
+
+        This will return either:
+        * "pending": The user has initiated a request, but no response has been
+            received from the credit provider.
+        * "approved": The user's credit request has been approved by the provider.
+        * "rejected": The user's credit request has been rejected by the provider.
+
         """
-        unique_together = ('username', 'course')
+        try:
+            return self.statuses.latest().status  # pylint: disable=no-member
+        except CreditRequestStatus.DoesNotExist:
+            return "pending"
+
+    @classmethod
+    def credit_requests_for_user(cls, username):
+        """
+        Retrieve all credit requests for a user.
+
+        Arguments:
+            username (unicode): The username of the user.
+
+        Returns: list
+
+        Example Usage:
+        >>> CreditRequest.credit_requests_for_user("bob")
+        [
+            {
+                "uuid": "557168d0f7664fe59097106c67c3f847",
+                "timestamp": "2015-05-04T20:57:57.987119+00:00",
+                "course_key": "course-v1:HogwartsX+Potions101+1T2015",
+                "provider": {
+                    "id": "HogwartsX",
+                    "display_name": "Hogwarts School of Witchcraft and Wizardry",
+                },
+                "status": "pending"  # or "approved" or "rejected"
+            }
+        ]
+
+        """
+        # To minimize the number of database queries, we execute
+        # the raw SQL query instead of using Django's ORM.
+        # This allows us to use a subquery to
+        # retrieve the most recent status for each credit request.
+        cursor = connection.cursor()
+        cursor.execute(
+            (
+                'SELECT cr.uuid, cr.created, cc.course_key, cp.provider_id, cp.display_name, '
+                'IFNULL(('
+                '    SELECT status FROM credit_creditrequeststatus AS crs '
```

Comment:
Use table names from the model's metadata (e.g. `CreditRequestStatus._meta.db_table`).